rotkehlchen/chain/evm/utils.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from typing import TYPE_CHECKING, Any, Callable, Final, Optional
+
+from web3.types import BlockIdentifier
+
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.errors.misc import BlockchainQueryError, RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import UNISWAP_PROTOCOL, VELODROME_POOL_PROTOCOL, EvmTokenKind, Price
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import LP_TOKEN_AS_POOL_CONTRACT_ABIS
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+LP_TOKEN_AS_POOL_PROTOCOL_TO_ABI_NAME: Final[dict[str, 'LP_TOKEN_AS_POOL_CONTRACT_ABIS']] = {
+    VELODROME_POOL_PROTOCOL: 'VELO_V2_LP',
+    UNISWAP_PROTOCOL: 'UNISWAP_V2_LP',
+}
+MULTICALL_DEPLOYMENT_BLOCK = 7929876
+
+
+def lp_price_from_pool_contract(
+        evm_inquirer: 'EvmNodeInquirer',
+        token: EvmToken,
+        token_price_func: Callable,
+        token_price_func_args: list[Any],
+        block_identifier: BlockIdentifier,
+) -> Optional[Price]:
+    """
+    This works for any uniswap like LP token. It calculates the price for an LP token the contract
+    of which is also the contract of the pool it represents. For example velodrome or uniswap lp
+    tokens. The price is calculated like this:
+    price = (Total value of liquidity pool) / (Current supply of LP tokens)
+    We need:
+    - Price of token 0
+    - Price of token 1
+    - Pooled amount of token 0
+    - Pooled amount of token 1
+    - Total supply of pool token
+    """
+    address = token.evm_address
+    try:
+        abi_name = LP_TOKEN_AS_POOL_PROTOCOL_TO_ABI_NAME[token.protocol]  # the lp token's contract abi that will be used to query the info needed for lp token price calculation  # noqa: E501
+    except KeyError:
+        log.debug(
+            f'There is no suitable contract abi for protocol {token.protocol} '
+            f'of token {token.evm_address}. Cannot calculate price',
+        )
+        return None
+
+    contract = EvmContract(
+        address=address,
+        abi=evm_inquirer.contracts.abi(abi_name),
+        deployed_block=0,
+    )
+    methods = ('token0', 'token1', 'totalSupply', 'getReserves', 'decimals')
+    if isinstance(block_identifier, int) and block_identifier <= MULTICALL_DEPLOYMENT_BLOCK:
```

Comment:
We thankfully keep deployed block attributes in `EvmContract` class! So easy fix here. Pheww!

```suggestion
    if isinstance(block_identifier, int) and block_identifier <= evm_inquirer.contract_multicall.deployed_block:
```